src/MoBi.Presentation/Presenter/Main/SimulationExplorerPresenter.cs

```diff
@@ -188,21 +188,51 @@ private void refreshDisplayedSimulation(IMoBiSimulation simulation)
          var simulationNode = _view.NodeById(simulation.Id);
          
          var changedTemplateBuildingBlocks = findChangedTemplateBuildingBlocks(simulation).ToList();
+         var changedModules = findChangedTemplateModules(simulation).ToList();
 
          var isChangedSimulation = changedTemplateBuildingBlocks.Any() || simulation.OriginalQuantityValues.Any();
          simulationNode.Icon = isChangedSimulation ? ApplicationIcons.SimulationRed : ApplicationIcons.SimulationGreen;
 
+         updateModuleIcons(changedModules.Select(x => x.simulationModule), simulationNode.AllNodes.OfType<ModuleConfigurationNode>());
          updateBuildingBlockIcons(changedTemplateBuildingBlocks.Select(x => x.simulationBuildingBlock), simulationNode.AllNodes.OfType<BuildingBlockNode>());
       }
 
```

Comment:
Would it make sense to move all of this stuff outside of the presenter? WE have all those tasks already. Does it belong in InteractionTaskForSimulation . Most of those methods are private . It's rare to have so much logic in a presenter